kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -319,32 +324,6 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
-    fn codegen_direct_literal(&mut self, ty: Ty<'tcx>, init: Expr) -> Expr {
-        let func_name = format!("gen-{}:direct", self.ty_mangled_name(ty));
-        let pretty_name = format!("init_direct::<{}>", self.ty_pretty_name(ty));
-        let cgt = self.codegen_ty(ty);
-        self.ensure(&func_name, |tcx, _| {
-            let target_ty = init.typ().clone(); // N
-            let param = tcx.gen_function_parameter(1, &func_name, target_ty);
-            let var = tcx.gen_function_local_variable(2, &func_name, cgt.clone()).to_expr();
-            let body = vec![
-                Stmt::decl(var.clone(), None, Location::none()),
-                tcx.codegen_discriminant_field(var.clone(), ty)
```

Comment:
`codegen_discriminant_field` has special handling for generators, which your code does not seem to cover. I don't think we should go back to hard-coding a "case" field everywhere.

Reply:
Added a check for this